Microsoft.Toolkit.Uwp.UI.Controls/Menu/MenuItem.cs

```diff
@@ -0,0 +1,152 @@
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Input;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// Menu Item is the items main container for Class Menu control
+    /// </summary>
+    public class MenuItem : ItemsControl
+    {
+        /// <summary>
+        /// MenuItem header text
+        /// </summary>
+        public static readonly DependencyProperty HeaderProperty = DependencyProperty.Register(nameof(Header), typeof(string), typeof(MenuItem), new PropertyMetadata(default(string)));
+
+        private Menu ParentMenu => this.FindAscendant<Menu>();
+
+        internal Button FlyoutButton { get; private set; }
+
+        /// <summary>
+        /// Gets a value indicating whether the menu is opened or not
+        /// </summary>
+        public bool IsOpened { get; private set; }
+
+        /// <summary>
+        /// Gets or sets the title to appear in the title bar
+        /// </summary>
+        public string Header
+        {
+            get { return (string)GetValue(HeaderProperty); }
+            set { SetValue(HeaderProperty, value); }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MenuItem"/> class.
+        /// </summary>
+        public MenuItem()
+        {
+            DefaultStyleKey = typeof(MenuItem);
+            IsFocusEngagementEnabled = true;
+        }
+
+        /// <summary>
+        /// This method is used to show the menu for current item
+        /// </summary>
+        public void ShowMenu()
+        {
+            FlyoutButton?.Flyout?.ShowAt(FlyoutButton);
+        }
+
+        /// <summary>
+        /// This method is used to hide the menu for current item
+        /// </summary>
+        public void HideMenu()
+        {
+            FlyoutButton?.Flyout?.Hide();
+        }
+
+        /// <inheritdoc />
+        protected override void OnApplyTemplate()
+        {
+            FlyoutButton = GetTemplateChild("FlyoutButton") as Button;
+            IsOpened = false;
+
+            if (FlyoutButton != null && Items != null && Items.Any())
+            {
+                var menuFlyout = new MenuFlyout();
+                menuFlyout.Placement = ParentMenu.Orientation == Orientation.Horizontal
+                    ? FlyoutPlacementMode.Bottom
+                        : FlyoutPlacementMode.Right;
+
+                menuFlyout.Opened -= MenuFlyout_Opened;
+                menuFlyout.Closed -= MenuFlyout_Closed;
+                FlyoutButton.PointerExited -= FlyoutButton_PointerExited;
+                menuFlyout.Opened += MenuFlyout_Opened;
+                menuFlyout.Closed += MenuFlyout_Closed;
+                FlyoutButton.PointerExited += FlyoutButton_PointerExited;
+
+                menuFlyout.MenuFlyoutPresenterStyle = ParentMenu.MenuFlyoutStyle;
+                FlyoutButton.Style = ParentMenu.HeaderButtonStyle;
+
+                foreach (var item in Items)
```

Comment:
What happens if the ItemSource changes during runtime, is there a way to do add/remove/move menuItem